common/modules/gcp-project/main.tf

```diff
@@ -0,0 +1,89 @@
+# this Terraform module creates the same resources as project_init script and
+# set the DNS zones structure.
+
+terraform {
+  backend "gcs" {}
+}
+
+variable "project_name" {}
+variable "project_owner" {}
+variable "billing_id" {}
+variable "organization_id" {}
+
+# compute the project_id to get the dns zone in JSON format
+# I wasn't able to replicate this function using HCL code
+data "external" "calculate_dns_zone" {
+  program = ["python", "-c", "print(\"{\\\"zone\\\": \\\"\" + \".\".join(\"${google_project.project.project_id}\".split(\"-\")[::-1]) + \".net.\") + \"\\\"}\""]
+}
+ 
+resource "google_project" "project" {
+  name            = "gpii-gcp-${var.project_name}"
+  project_id      = "gpii-gcp-${var.project_name}"
+  billing_account = "${var.billing_id}"
+  org_id          = "${var.organization_id}"
+}
+
+resource "google_project_services" "project" {
+  project = "${google_project.project.project_id}"
+  services = [
+    "compute.googleapis.com",
+    "logging.googleapis.com",
+    "monitoring.googleapis.com",
+    "dns.googleapis.com"
+  ]
+}
+
+resource "google_service_account" "project" {
+  account_id   = "projectowner"
+  display_name = "Project owner service account"
+  project      = "${google_project.project.project_id}"
+}
+
+resource "google_project_iam_binding" "project" {
+  project = "${google_project.project.project_id}"
+  role    = "roles/owner"
+  members = [
+    "user:${var.project_owner}",
+    "serviceAccount:${google_service_account.project.email}",
+  ]
+}
+ 
+resource "google_dns_managed_zone" "project" {
```

Comment:
My first attempt would be to import the resource before running the `up` command in order to be "ignored" in the apply step. This could be the less disruptive change in the infrastructure. 

Other approaches could be either removing the resource from the gke-network module, or change the zone creation at this PR.

But I can't either test or evaluate the impact all of the above before deploying the project first.

Reply:
Ok, I did some testing. Looks like the easiest solution that does not require any changes to Terraform code – do not set dns records and zones here:
https://github.com/gpii-ops/gpii-infra/blob/master/gcp/rakefiles/vars.rb#L37